net/FlatBuffers/FlatBufferBuilder.cs

```diff
@@ -226,15 +230,35 @@ public void NotNested()
         {
             // You should not be creating any other objects or strings/vectors
             // while an object is being constructed
-            if (_vtable != null)
+            if (_vtableSize >= 0)
                 throw new Exception(
                     "FlatBuffers: object serialization must not be nested.");
         }
 
         public void StartObject(int numfields)
         {
+            if (numfields < 0)
+                throw new ArgumentOutOfRangeException("Flatbuffers: invalid numfields");
+
             NotNested();
-            _vtable = new int[numfields];
+
+            if (_vtable.Length < numfields)
+            {
```

Comment:
This seems overly complex code for no reason. It be way simpler to simply set the new length to the size of the current table. Powers of two are in theory powers of two cause less allocations, but in practice this difference is minimal.